### PR TITLE
style(menu): clean up right menu - leaving only "Settings"

### PR DESCRIPTION
**Pull request by mistercrunch**
### SUMMARY
- removing the big `+ NEW` button in the navbar as we have redundant NEW buttons in our easily accessible list views
- hiding i18n flags selector by default as most translations are assumed to be incomplete
- merging the User menu under settings

### after
Screen Shot 2020-10-10 at 1 15 22 AM

### before
Screen Shot 2020-10-10 at 10 45 57 AM
Screen Shot 2020-10-10 at 10 47 36 AM
Screen Shot 2020-10-10 at 10 47 28 AM

### CODE REVIEW

File: superset/views/utils.py [hunk 1]
```diff
@@ -351,11 +351,11 @@ def build_extra_filters(  # pylint: disable=too-many-locals,too-many-nested-bloc
     for filter_id, columns in default_filters.items():
         filter_slice = db.session.query(Slice).filter_by(id=filter_id).one_or_none()
 
-        filter_configs = (
-            json.loads(filter_slice.params or "{}").get("filter_configs") or []
-            if filter_slice
-            else []
-        )
+        filter_configs: List[Dict[str, Any]] = []
```
Comment (mistercrunch) [inline superset/views/utils.py:354]: Somehow `mypy` + `pre-commit` were picking up on this, unclear why as i didn't touch the file. Added type annotations here to move forward...

**Comment by CoryChaplin** [comment]
🤔 I only use the big new button, going into the charts/dashboard/lab sections to do a new thing feels like an extra click.
Do you have insights about current usage for creating new items?

**Comment by mistercrunch** [comment]
We may keep the big `+ NEW` button, asking designers for input, let's see what they think.

**Comment by mistercrunch** [comment]
One note is that `ENABLE_REACT_CRUD_VIEWS` is still set to `False` by default, and we may want to flip that to `True` prior to removing the `+ NEW` button.

**Comment by mistercrunch** [comment]
Addressing comments from my last comment here https://github.com/apache/incubator-superset/pull/11259

**Review by pkdotson — APPROVED** [review]
LGTM! New menu looks pretty neat!

**Comment by suddjian** [comment]
We should definitely turn on React CRUD by default before making this change.

Another thing to consider: people seem to use the `+ NEW` button a lot, and I think many would be unhappy needing to navigate to an extra page before they can create something. I wonder if it would make more sense instead to do a visual redesign of it so that it's less prominent. Many products have similar buttons:

Screen Shot 2020-10-20 at 9 24 12 AM

Screen Shot 2020-10-20 at 9 23 02 AM

Screen Shot 2020-10-20 at 9 40 53 AM

**Comment by mistercrunch** [comment]
Love the idea @suddjian , let me try a subtle `+`

**Comment by mistercrunch** [comment]
Screen Shot 2020-10-20 at 12 22 38 PM

**Comment by mistercrunch** [comment]
@suddjian what do you think? ^^^

**Review by suddjian — APPROVED** [review]
Love it!

**Pull request merged**